In [1]:
import numpy as np
import pandas as pd
from sklearn.model_selection import cross_val_score,KFold
from sklearn.pipeline import Pipeline
from sklearn.ensemble import RandomForestRegressor,GradientBoostingRegressor
from sklearn.metrics import mean_squared_error
from sklearn.tree import DecisionTreeRegressor
from xgboost import XGBRegressor
from lightgbm import LGBMRegressor
from sklearn.linear_model import ElasticNet,Ridge,Lasso
from sklearn.preprocessing import StandardScaler

In [2]:
X_train = pd.read_parquet("../data/processed/train_X_fe.parquet")
y_train = pd.read_parquet("../data/processed/train_y.parquet").squeeze()

# Base Models

In [3]:
models = {
    "DecisionTree": DecisionTreeRegressor(random_state=42),
    "RandomForest": RandomForestRegressor(random_state=42, n_jobs=-1),
    "XGBoost": XGBRegressor(random_state=42, n_jobs=-1),
    "LightGBM": LGBMRegressor(random_state=42),
    "GBR": GradientBoostingRegressor(random_state=42),
    "ElasticNet": Pipeline([("scaler", StandardScaler()), ("model", ElasticNet(random_state=42))]),
    "Lasso": Pipeline([("scaler", StandardScaler()), ("model", Lasso(random_state=42))]),
    "Ridge": Pipeline([("scaler", StandardScaler()), ("model", Ridge())])
}

In [4]:
for name, model in models.items():
    scores = cross_val_score(model, X_train, y_train, cv=5, scoring="neg_root_mean_squared_error")
    print(f"{name} CV RMSE (on log-scale) : {-scores.mean()}")

DecisionTree CV RMSE (on log-scale) : 0.21716530917047994
RandomForest CV RMSE (on log-scale) : 0.13733375313565585
XGBoost CV RMSE (on log-scale) : 0.13745868645611709
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Auto-choosing col-wise multi-threading, the overhead of testing was 0.002129 seconds.
You can set `force_col_wise=true` to remove the overhead.
[LightGBM] [Info] Total Bins 2274
[LightGBM] [Info] Number of data points in the train set: 990, number of used features: 133
[LightGBM] [Info] Start training from score 12.027392
[LightGBM] [Warning] Found whitespace in feature_names, replace with underlines
[LightGBM] [Info] Auto-choosing row-wise multi-threading, the overhead of testing was 0.000993 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 2280
[LightGBM] [Info] Number of data points in the train set: 990, number

# Final Verdict

1. **RandomForest** CV RMSE (on log-scale) : 0.13733375313565585
2. **XGBoost** CV RMSE (on log-scale) : 0.13745868645611709
3. **LightGBM** CV RMSE (on log-scale) : 0.13136620147730904
4. **Ridge** CV RMSE (on log-scale) : 0.12219688124396784